chapters/location_component_interface.adoc

```diff
@@ -0,0 +1,98 @@
+// Copyright 205 The Khronos Group, Inc.
+// SPDX-License-Identifier: CC-BY-4.0
+
+ifndef::chapters[:chapters:]
+ifndef::images[:images: images/]
+
+[[location-and-component-interface]]
+= Location and Component Interface
+
+This chapter is an overview of the link:https://docs.vulkan.org/spec/latest/chapters/interfaces.html#interfaces-iointerfaces-locations[Location and Component Assignment] chapter to help give examples, especially around some of the more extreme edge cases.
+
+The simplest way to think about a `Location` is that it is made up of four 32-bit `Components`. This means a `vec4`/`float4`/`uvec4`/etc will fit perfectly in a single `Location`.
+
+`Location` are used for both the `Input` and `Output` to interface between shaders stages when possible.
+
+== Basic Example
+
+Here is a basic example:
+
+[source,glsl]
+----
+layout(location=0) in vec4 a;
+layout(location=1, component = 0) in vec2 b;
+layout(location=1, component = 2) in float c;
+----
+
+image::{images}location_example_basic.svg[location_example_basic]
+
+== 16-bit
+
+When using 16-bit value, they consume a full 32-bit `Component` for each 16-bit element. So a 16-bit wide vector will look the same as a 32-bit vector.
+
+[source,glsl]
+----
+layout(location=0) in f16vec3 a;
+----
+
+image::{images}location_example_16bit.svg[location_example_16bit]
+
+== Crossing Location Boundaries
+
+All 16-bit and 32-bit vectors must be inside a single `Location`, so the following in **not** allowed.
+
+[source,glsl]
+----
+layout(location=0, component = 2) in vec4 a;
+----
+
+image::{images}location_example_boundaries.svg[location_example_boundaries]
+
+== 64-bit
+
+64-bit are special as they can consume up to 2 `Location`, but they must only start at `Component` `0` or `2`.
+
+[source,glsl]
+----
+layout(location=0) in f64vec3 a;
+----
+
+image::{images}location_example_64bit.svg[location_example_64bit]
+
+== Interleaving Components
+
+The following attempt to have multiple variables alias the same component is link:https://godbolt.org/z/h61baYhT4[not allowed].
+
+[source,glsl]
+----
+layout(location=0) in vec2 in_a;
+layout(location=0, component=1) in float in_f;
+----
+
+== Array
+
+Arrays will consume the entire `Location` for a given element in the array, as an example
+
+[source,glsl]
+----
+layout(location=0, component=1) in float a[3];
+----
+
+image::{images}location_example_array.svg[location_example_array]
+
+As seen, using a scalar or something such as a `vec2`/`float2` will leave many `Component` slots unused.
+
+It is **not** allowed to use any other `Component` in a `Location` that is being consumed by an array
+
+[source,glsl]
+----
+layout(location=0, component=1) in float a[3];
+layout(location=2, component=2) in float b;
```

Comment:
A comment here explaining would be helpful - could also be a sentence afterwards.

```suggestion
layout(location=2, component=2) in float b;  // Invalid, as the array consumes all of location 2.
```